chore(aws): add irsa documentation for next

Langleu opened the pull request:
## Description

<!-- This helps the reviewers by providing an overview of what to expect in the PR. Linking to an issue is preferred but not required. -->
<!-- Add an assignee and use component: labels for good hygiene! -->

Adds documentation related to AWS's IAM Roles for Service Accounts (IRSA) feature.
More prominent in the next release.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manage, tech lead, or infra engineer as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

Langleu commented:
Hey @aabouzaid, would love to get a review from your side since you're also maintaining the helm chart.
@christinaausley,  would love to get a technical writer review since I don't have that much public docs experience.

christinaausley commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
````diff
@@ -0,0 +1,327 @@
+  namespace: aurora-namespace
+```
+
+The setup required on the Aurora Postgres side is to actually create the user and assign the required permissions to it. The following is an example when connected to the Postgres database, and can also be realized differently. See the [AWS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html#UsingWithRDS.IAMDBAuth.DBAccounts.PostgreSQ ...
````
What do you mean by "realized differently" here?

christinaausley commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,327 @@
+
+Required are the following artifacts for the `aws-advanced-jdbc-wrapper` to work:
+
+- https://mvnrepository.com/artifact/software.amazon.awssdk/regions
```
I'm trying to think of a way to make this more clean. Could we instead provide [words]() with the links supplied to trim it up?

Langleu replied:
good idea! I'll switch it up to just be the artifact name and also mention prior that all of those artifacts are from `software.amazon.awssdk` just to make sure there's no misunderstanding.

christinaausley commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,327 @@
+
+### Instance Metadata Service (IMDS)
+
+[IMDS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) is a default fallback for the AWS SDK. Within the context of EKS, it means a pod will automatically assume the role of a node. This can hide many problems, including whether IRSA was set up correctly or not, since it will fall back to IMDS in case of failure and hide the actual error.
```
I would spell IMDS out on first reference here for clarity.

christinaausley commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
````diff
@@ -0,0 +1,327 @@
+        http_put_response_hop_limit = 1
+    }
+}
+```
````
Any additional resources or next steps you would provide here?

Langleu replied:
I think it will be a living document and grow when released to the public.
Based on future help requests we can adjust this.
For now, it's just an indicator for users to disable to help debug the issue as otherwise this system will hide the actual error and make debugging impossible.

christinaausley reviewed (changes requested):
Nice work here! Technical review in latest commit, and a few notes for suggestions:

- See comments inline.
- We have some large chunks of text, is it possible to break any of these sections into smaller ones, or into a series of steps rather than just text?
- Are there any places here you might suggest screenshots to help direct users? This could also help break up the chunks of text.

Langleu commented:
thanks @christinaausley for the quick review, I really appreciate it!
I'll have a look at your suggestions and on splitting things up / adding screenshots and will let you know when I'm done!

aabouzaid commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,327 @@
+Where the value is whatever the endpoint of your OpenSearch cluster is.
+
+:::note
+AWS OpenSearch runs generally on port 443.
```
I don't know if it's only me, but for a second, the word `run` confused me even its technically valid :sweat_smile: 
Maybe "listen" more a clear word :thinking:

Langleu replied:
rephrased it to `listens` and added that it usually runs on 9200, so people may better understand why it's noted.

aabouzaid commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
````diff
@@ -0,0 +1,327 @@
+
+As an example, configure the following environment variables
+
+```
````
Add `yaml` for a better format.

aabouzaid commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
````diff
@@ -0,0 +1,327 @@
+
+Using a Terraform module like the [EKS module](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest), one can define the following to decrease the default value of two to one, which results in pods not being allowed to assume the role of the node anymore.
+
+```
````
Add `json` for consistency with other HCL sections.

Langleu replied:
ah thanks, was wondering what those are using, awesome!

aabouzaid replied:
In the end, HCL is as ugly as JSON 😂

Langleu commented:
Hey @christinaausley, I adjusted the requested changes.
I don't have good ideas for screenshots since it's very configuration focused. Therefore, added some sub headers for each chapter, which can be used for referencing.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,327 @@
+
+### Keycloak
+
+For Keycloak versions 21+, the default JDBC driver can be overwritten, allowing use of a custom wrapper like the [aws-advanced-jdbc-wrapper](https://github.com/awslabs/aws-advanced-jdbc-wrapper) to utilize the features of IRSA. This is a wrapper around the default JDBC driver, but takes care of signing the requests.
```
on v19 it simply doesn't work 😆.
Really only from v21 onwards it will work, nothing below that is supported to change the jdbc driver.

aabouzaid replied:
In that case, we can explicitly mention it's only supported starting from v21 so it's clear why we don't cover older versions in this guide.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,327 @@
+description: "Learn how to configure IAM roles for service accounts (IRSA) within AWS to authenticate workloads."
+---
+
+IAM roles for service accounts (IRSA) is a way within AWS to authenticate workloads in EKS (Kubernetes), for example, to execute signed requests against AWS services. This is a replacement for basic auth and is generally considered more secure.
```
rephrased it slightly and linked it to the best practices recommendation by them.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,327 @@
+
+## Aurora Postgres
+
+When using the Terraform provider of [AWS](https://registry.terraform.io/providers/hashicorp/aws/latest) with the resource [aws_rds_cluster](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster) to create a new rational database (RDS) or Aurora cluster, supply the argument `iam_database_authentication_enabled = true` to enable the IAM roles functionality. See th ...
```
I added a section under "troubleshooting" that lists all the versions used within the docs and a disclaimer that it may work with newer versions as well.

Langleu commented:
I resolved all mentioned requested changes.
Let me know if there's anything else!
I appreciate the work you've both put into reviewing this 🙇

Langleu commented:
Hey @christinaausley, indeed! 🚀

aabouzaid commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/irsa.md:
```diff
@@ -0,0 +1,390 @@
+| ----------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
+| AWS Aurora PostgreSQL                                                                                                                                 | 13 / 14 / 15 |
+| AWS OpenSearch                                                                                                                                        | 2.5          |
+| [Aws-advanced-jdbc-wrapper](https://github.com/awslabs/aws-advanced-jdbc-wrapper)                                                                     | 2.2.2        |
```
Let's use a normal name like the rest of the items in the list here.

So it will be: [Amazon Web Services (AWS) JDBC Driver](https://github.com/awslabs/aws-advanced-jdbc-wrapper)

aabouzaid reviewed (changes requested):
Everything looks great, just 2 minor comments.
Thanks Lars :bow:

christinaausley reviewed (approved):
Technical review in latest commit 👍 Really nice perseverance through this in resolving suggestions. Great work.

akeller commented:
@Langleu if you are happy with this PR, feel free to merge since it looks like you've addressed feedback from your reviewers.

Langleu commented:
Thank you, indeed! Didn't come around to do it yesterday.
Thanks everyone 🚀
